### ISSUE TITLE
No consistent, documented Autoconf formatting style for Expat's configure.ac

### ISSUE THREAD

**Issue by hartwork** [libexpat/libexpat#306]
Pull request #301 made me aware that we have very inconsistent formatting in `configure.ac` and no clear rules how `configure.ac` should be formatted.

While not 100% consistent, some existing formatting in our `configure.ac` — mostly contributed by @marcomaggi — is beautiful (and so are some bits presented at [Autoconf Conding Style](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Coding-Style.html#Coding-Style) as well as [The Autoconf Language](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Autoconf-Language.html)), despite (and probably also *because*) some unusual aspects of the approach taken.
What I'm trying to do here is to distill a *small* set of rules from the current style, add some more consistency to it, and put it in stone as a reference. After that, we can refer to this ticket in a pull request and say something like:

> This does not fit any of the variants proposed in formatting rule (2). Please apply 2a, 2b or 2c — as you prefer.

@marcomaggi do you feel like the core of your approach to Autoconf formatting can still be found in the rules below? @kkkunche does this look acceptable to you?

# Version 2019.08.09.21.54

## 1. Quoting
Quote "everything":
```
AC_DEFINE([HAVE_FOO], [1], [Define to 1 if you have the `foo' function.])
```

## 2. Parameter indentation

Parameters to functions either go
- (a) on the the same line or
- (b) align vertically or
- (c) go to the next line, with the first character indented 2 spaces more than the first *non-`[`*(!) parent level character, i.e. 2 or [3 columns further right](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Autoconf-Language.html):

```
CALL([parameter], [parameter], [parameter])

CALL([parameter],
     [parameter],
     [parameter])

CALL([parameter], [parameter],
  [CALL(
     [CALL()])])

  ^  ^
  |  2 + 3(!) spaces
  2 spaces
```

## 3. Consecutive call / multi-line indentation

Consecutive calls to macros (= on the the same nesting level) are aligned vertically:

```
CALL([parameter],
  [CALL([])
   CALL([])
   CALL([])])
```

## 4. Closing bracket placement
Closing braces accumulate on the same line in general...

```
CALL(
  [CALL([CALL([])],
        [CALL([])])
   CALL([])])
```

...but can go a new line (e.g. with `AC_LANG_SOURCE`) to match parameter indentation rule (2), i.e. either

```
CALL([CALL([
    one
    two
  ])],
  [CALL()])
```

.. or ..
```
CALL([CALL([
        one
        two
     ])],
     [CALL()])
```

**Comment by kkkunche** [comment]
Thanks for creating certain rules to have  Autoconf formatting style for configure.ac.  This is really important. When i made the changes in configure.ac, my reference was already existing code  which i assumed it is indentation correct hence i had a big confusion with the review comments you gave.

**Comment by kkkunche** [comment]
Done some changes related to indentation with pull request #307

**Comment by kkkunche** [comment]
@hartwork : Can we close this ?. Are there any pending items to work on this. Please let me know i can fix  it. Thanks

**Comment by hartwork** [comment]
I'd think we can close this issue once people editing configure.ac have a better chance at finding our style guide. Either we put it in a file or we link to this ticket for reference. Former would have the benefit that it works when GitHub is down. Maybe `configure-ac-style.md` in Git?

**Comment by hartwork** [comment]
Fixed by #315, closing.
